bittensor/_neuron/text/core_validator/__init__.py

```diff
@@ -606,24 +610,31 @@ def run_epoch( self ):
         self.prometheus_gauges.labels("dividends").set( self.metagraph.dividends[self.uid] )
         self.prometheus_gauges.labels("emission").set( self.metagraph.emission[self.uid] )
 
+        # Iterate epochs.
+        self.epoch += 1
+
     def metagraph_sync(self):
         r""" Syncing metagraph together with other metagraph-size related objects
         """
-        old_hotkeys = self.neuron_hotkeys if self.neuron_hotkeys else self.metagraph.hotkeys
-        self.metagraph.sync()
+        old_hotkeys = self.neuron_hotkeys + [] if self.neuron_hotkeys else self.metagraph.hotkeys
```

Comment:
what does adding an empty list to the back of the old hotkeys do?

Reply:
Just a sanity measure to ensure `old_hotkeys` is not a reference to a list that will be modified outside, but is rather a copy of the list, which is what `+ []` does.